### ISSUE TITLE
💅 noExtraNonNullAssertion false positive with two ! on different variables

### ISSUE THREAD

**Issue by sparr** [biomejs/biome#8314]
### Environment information

```bash
(possibly inaccurate due to running biome as vscode extension)

CLI:
  Version:                      2.3.8
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.7.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Not set
  Path:                         unset

Workspace:
  Open Documents:               0
```

### Rule name

lint/suspicious/noExtraNonNullAssertion

### Playground link

https://biomejs.dev/playground/?lintRules=noExtraNonNullAssertion&code=YQAhACsAPQBiACEAOwAKAA%3D%3D&language=ts

### Expected result

```ts
a!+=b!;
```

This rule should not trigger when the two `!` are being applied to different values.

(noNonNullAssertion would still trigger twice, but that rule might be separately disabled)

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ghost** [comment]
Below is part of the `noExtraNonNullAssertion` implementation that flags cases where it encounters a `TsNonNullAssertionExpression` on the right-hand side of a `JsAssignmentExpression`, and its left-hand side is of type `TsNonNullAssertionAssignment`, such as `a! += b!`:

https://github.com/biomejs/biome/blob/8de2774fb507a10e32ecf920bb5d0f801a9e869c/crates/biome_js_analyze/src/lint/suspicious/no_extra_non_null_assertion.rs#L98-L103

Simply removing this match arm appears to resolve the issue without breaking any existing test cases for this rule. Should I submit a PR for this change?

**Comment by DayKev** [comment]
Seems like it was fixed by #8477, the playground link doesn't emit any diagnostics now.
